geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/command/ExecuteFunction66.java

```diff
@@ -330,7 +330,8 @@ private void executeFunctionaLocally(final Function fn, final FunctionContext cx
 
     if (fn.hasResult()) {
       fn.execute(cx);
-      if (!((ServerToClientFunctionResultSender65) sender).isLastResultReceived()
+      if (sender.isOkayToSendResult()
+          && !((ServerToClientFunctionResultSender65) sender).isLastResultReceived()
```

Comment:
Can this cast be removed? It looks like sender is already this type

Reply:
@dschneider-pivotal i will remove this as part of another check in. I will have to re-run tests to confirm that removing  this is not an issue.